src/artifacts.js

```diff
@@ -176,7 +220,31 @@ api.declare({
   var isPublic = /^public\//.test(name);
   var details  = {};
   switch (storageType) {
```

Comment:
@jhford, I feel I must apologizes for the horrors that is the artifact API end-point implementation... It's so messy...

If you 110% fired up feel free to refactor all of this into different functions... perhaps it would have been smartest to just start the this handler with:
```js
switch (storageType) {
  case 'blob':
    return createAritfactBlob(req, res);
  case ...
}
```

and have separate methods for each artifact type... I know it would imply some code duplication, but right now we have sprinkled special-cases all over the place :)